src/server/managers/AppListenerManager.ts

```diff
@@ -618,6 +620,23 @@ export class AppListenerManager {
         }
     }
 
+    private async executePreRoomUserJoined(data: IPreRoomUserJoinedContext): Promise<void> {
+        data.room = new Room(data.room, this.manager);
```

Comment:
The `Utilities.deepCloneAndFreeze` isn't necessary here?

Reply:
No - the `data` object is created by Rocket.Chat only to serve the purpose of this event. No objects inside it are referenced out of the bridges; so it wouldn't really matter if the app changes it

Reply:
Thinking about it now, though, if an app does change the room object it might have an impact on other apps that are listening for the same event.